indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -79,6 +79,8 @@
   private static final int NUM_RETRIES = 8;
   private static final int SECONDS_BETWEEN_RETRIES = 2;
   private static final int DEFAULT_FS_BUFFER_SIZE = 1 << 18; // 256KB
+  private static final Pattern SNAPSHOT_JAR = Pattern.compile(".*SNAPSHOT(-selfcontained)?\\.jar$");
```

Comment:
nit: this should be
```
Pattern.compile(".*\\-SNAPSHOT(-selfcontained)?\\.jar$")
```
just in case someone was crazy enough to have artifact name be something like iAmNotSNAPSHOT.jar , it is not a "snapshot" jar really.